File: February 27, 2026.py
def shift_matrix(matrix, shift):

    new_matrix = []
    arr = []
    m = len(matrix)
    n = len(matrix[0])
    count = 0

    for i in range(m):
        for j in range(n):
            arr.append(matrix[i][j])

    l = len(arr)
    shift = ((shift % l) + l) % l

    for i in range(shift):
        arr.insert(0, arr.pop())

    for i in range(m):
        row = []
        for j in range(n):
            row.append(arr[count])
            count += 1
        new_matrix.append(row)

    return new_matrix

File: February 27, 2026.js
function shiftMatrix(matrix, shift) {

  let newMatrix = [];
  let arr = [];
  let m = matrix.length;
  let n = matrix[0].length; 
  let count = 0;

  for(let i = 0; i < matrix.length; i++) {
    for(let j = 0; j < matrix[i].length; j++) {
      arr.push(matrix[i][j])
    }
  }

  let l = arr.length;
  shift = ((shift % l) + l) % l;

  for (let i = 0; i < shift; i++){
    arr.unshift(arr.pop());
  }
    
  for(let i = 0; i < m; i++) {

    newMatrix[i] = []

    for(let j = 0; j < n; j++) {
      newMatrix[i][j] = arr[count]
      count++;
    }
  }

  return newMatrix;
}